Resources/public/js/views/services/ez-navigationhubviewservice.js

```diff
@@ -53,6 +54,31 @@ YUI.add('ez-navigationhubviewservice', function (Y) {
             });
         },
 
+        /**
+         * activeNavigationChange event handler, it redirects to the first item of the clicked zone
+         *
+         * @method _activeNavigationChange
+         * @protected
+         * @param {EventFacade} e
+         * @param {String} e.newVal new value of `activeNavigation`
+         */
+        _activeNavigationChange: function (e) {
+            var zone = this.get(e.newVal + 'NavigationItems'),
+                firstItem,
+                route;
+
+            if (zone && zone.length > 0) {
+                firstItem = zone[0];
+
+                if (firstItem.config) {
```

Comment:
we already have the same kind of test to detect whether it's a config or if we already have a true navigation item view and this is done by checking the `get` method in https://github.com/ezsystems/PlatformUIBundle/blob/master/Resources/public/js/views/services/ez-navigationhubviewservice.js#L343 and https://github.com/ezsystems/PlatformUIBundle/blob/master/Resources/public/js/views/services/ez-navigationhubviewservice.js#L319 so I guess it's time to introduce a small private method to do that.